### PR TITLE
Add support for chrome MediaSession API

### PR DESCRIPTION
**Pull request by chrisrhoden**
Basically makes playback control work nicer on android devices, it is a standard which
will hopefully pick up in other places as well.

### CODE REVIEW

**Comment by chrisrhoden** [comment]
Before:
screen shot 2017-07-14 at 11 04 42 am

After:
screen shot 2017-07-14 at 11 04 55 am

**Comment by gcampo88** [comment]
dang it. watch out for rickrolling in the docs!

File: src/app/shared/player/player.component.ts [hunk 1]
```diff
@@ -268,7 +274,7 @@ export class PlayerComponent implements OnInit, OnChanges {
 
   togglePlayPause() {
     if (this.player.paused) {
-      this.player.play();
+      this.player.play().then(_ => this.sessionService.playbackStarted());
```
Comment (chrisrhoden) [inline src/app/shared/player/player.component.ts]: It's all pulled automatically by the browser from whether there is an audio element playing on the page. We put this in place to not show the notification until audio playback has started, so if there's an error a notification doesn't show up anyway.
The API for mediasession is literally just metadata and callbacks for transport controls, so you can't even report a play state.

File: src/app/shared/player/mediasession.service.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        this.album = album;
+        this.albumArtUrl = albumArtUrl;
+        if (this.triggered) {
+            this.playbackStarted();
+        }
+    }
+    playbackStarted() {
+        this.triggered = true;
+        if (this.title && 'mediaSession' in navigator) {
+            console.log('media session ')
```
Comment (chrisrhoden) [inline src/app/shared/player/mediasession.service.ts]: lemme remove

**Comment by cavis** [comment]
You had me at "remove debug log".

**Pull request merged**